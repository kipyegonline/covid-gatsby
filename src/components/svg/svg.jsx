import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

//svg
export const SVG = styled.svg.attrs(props => ({
  height: props.h,
  width: props.w,
  className: props.classlist,
}))`
  background: ${props => props.bg};
  border-radius: 10px;
  margin: auto;
`

SVG.propTypes = {
  h: PropTypes.number.isRequired,
  w: PropTypes.number.isRequired,
  bg: PropTypes.string.isRequired,
}
SVG.defaultProps = {
  h: 480,
  w: 500,
  bg: "#fefefe",
}
//group tag
export const Group = styled.g.attrs(props => ({
  className: props.classlist,
  height: props.gh,
  width: props.gw,
}))`
  transform: translate(${props => props.x}px, ${props => props.y}px);
`
Group.propTypes = {
  classlist: PropTypes.string,
  gh: PropTypes.number,
  gw: PropTypes.number,
  x: PropTypes.number,
  y: PropTypes.number,
}
Group.defaultProps = {
  gw: 0,
  gh: 0,
  classlist: "",
  x: 0,
  y: 0,
}
//rect
export const Rect = styled.rect.attrs(props => ({
  onClick: props.hc,
  onMouseOver: props.hh,
  onMouseLeave: props.hl,
  className: props.classlist,
  x: props.x,
  y: props.y,
  width: props.width,
  height: props.height,
}))`
  shaperendering: crispEdges;
  opacity: ${props => props.opacity};
  fill: ${props => props.fill};
  transform: translate(${props => props.tx}px, ${props => props.ty}px);
  transition: all 0.2s linear;

  &:hover {
    fill: orange;
    width: ${props => props.hover + 5}px;
    opacity: ${props => props.opacitad};
    z-index: 10;
  }
`
Rect.propTypes = {
  hc: PropTypes.func,
  hh: PropTypes.func,
  hl: PropTypes.func,
  classlist: PropTypes.string,
  x: PropTypes.number,
  y: PropTypes.number,
  width: PropTypes.number,
  height: PropTypes.number,
  opacity: PropTypes.number,
  ty: PropTypes.number,
  tx: PropTypes.number,
  hover: PropTypes.number,
}
Rect.defaultProps = {
  x: 0,
  y: 480,
  height: 0,
  tx: 0,
  ty: 0,
}
//Text
export const Text = ({ x, y, text, transform, classlist }) => (
  <text x={x} y={y} fill={"black"} transform={transform} className={classlist}>
    {text}
  </text>
)
;`
   x:${props => props.x};
   y:${props => props.y};
   fill:${props => props.fill};
   `

//path

export const Path = styled.path.attrs(props => ({
  onClick: props.hc,
  onMouseOver: props.hh,
  onMouseLeave: props.hl,
  className: props.classlist,
  d: props.d,
}))`
  opacity: ${props => props.opacity};
  stroke: ${props => props.stroke};
  fill: ${props => props.fill};
  shaperendering: crispEdges;
  strokewidth: ${props => props.strokew}px;
`

Path.propTypes = {
  hc: PropTypes.func,
  hh: PropTypes.func,
  hl: PropTypes.func,
  classlist: PropTypes.string,
  d: PropTypes.string,
  opacity: PropTypes.number,
  stroke: PropTypes.string,
  fill: PropTypes.string,
  strokew: PropTypes.number,
}
//circle
export const Circle = props => (
  <circle
    cx={props.cx}
    cy={props.cy}
    r={props.r}
    shapeRendering="crispEdges"
    opacity={props.opacity}
    fill={props.fill}
    onClick={props.hc}
    onMouseOver={props.hh}
    onMouseLeave={props.hl}
  />
)

export const Circler = styled.circle.attrs(props => ({
  onClick: props.hc,
  onMouseOver: props.hh,
  onMouseLeave: props.hl,
}))`
  cx: ${props => props.cx}px;
  cy: ${props => props.cy}px;
  r: ${props => props.r}px;
  shaperendering: crispEdges;
  fill: ${props => props.fill};
  opacity: ${props => props.opacity};
  transition: all.2s easse-in-out;
  &:hover {
    fill: deepskyblue;
  }
`
Circle.propTypes = {
  hc: PropTypes.func,
  hh: PropTypes.func,
  hl: PropTypes.func,
  classlist: PropTypes.string,
  cx: PropTypes.number.isRequired,
  cy: PropTypes.number.isRequired,
  r: PropTypes.number.isRequired,
  fill: PropTypes.string,
  opacity: PropTypes.number,
}

Circle.defaultProps = {
  cx: 0,
  cy: 0,
  fill: "black",
  r: 5,
  opacity: 0.8,
}
export const ToolTip = styled.div.attrs(props => ({
  className: "covid-tip",
}))`
  width: ${props => props.data.w || 0}px;
  border-radius: 10px;
  background: ${props => props.data.bg};
  padding: 0.5rem;
  font-size: 1rem;
  margin: 5px;
  top: ${props => props.data.y / 2 || 0}px;
  left: ${props => props.data.x + 40 || 0}px;
  position: absolute;
  z-index: 110;
  height: auto;
  opacity: ${props => (props.tool ? 1 : 0)};
  transition: all 0.25s ease-in;
  box-shadow: -2px -2px 3px #fff, 2px 2px 3px #fff;
  pointer-events: none;

  @media (max-width: 480px) {
     {
      padding: 0.5rem;
      text-align: left !important;

      margin: 5px;
      width: ${props => props.data.w / 2 || 0}px;
      font-size: 1rem;
    }
  }
  @media (min-width: 480px) and(max-width:768px) {
     {
      padding: 0.5rem;
      text-align: left !important;
      margin: 5px;
      width: ${props => props.data.w / 2 || 0}px;
      font-size: 1rem;
    }
    & img {
      width: 30px;
    }
  }
`
ToolTip.propTypes = {
  w: PropTypes.number,
  bg: PropTypes.string,
  y: PropTypes.number,
  x: PropTypes.number,
  tool: PropTypes.bool,
}
ToolTip.defaultProps = {
  w: 100,
  bg: "#fff",
  tool: false,
  x: 0,
  y: 0,
}
export const Tip = styled.span.attrs(props => ({
  className: props.classlist,
}))`
  padding: 0.25rem;
  display: block;
  margin: 0;
  font-size: 1rem;
  font-family: roboto;
  @media (max-width: 480px) {
    padding: 0.25rem;
    font-size: 1rem;
  }
`

Tip.propTypes = {
  classlist: PropTypes.string,
}
export const TextArea = ({
  children,
  x,
  y,
  fill,
  rotate,
  fontSize,
  fontWeight,
}) => (
  <text
    x={x}
    y={y}
    fill={fill}
    fontWeight={fontWeight}
    transform={`rotate(${rotate})`}
    fontSize={fontSize}
  >
    {children}
  </text>
)

export const setLables = (x, y, fill, rotate, fontSize, fontWeight) => {
  const w = globalThis.window && document.documentElement.clientWidth
  if (w <= 480) {
  } else if (w <= 768) {
  } else {
    return { x, y, fill, rotate, fontSize, fontWeight }
  }
  return { x, y, fill, rotate, fontSize, fontWeight }
}

TextArea.propTypes = {
  children: PropTypes.node,
  fill: PropTypes.string,
  rotate: PropTypes.number,
  x: PropTypes.number,
  y: PropTypes.number,
  fontSize: PropTypes.string,
}

TextArea.defaultProps = {
  fill: "black",
  rotate: 0,
  x: 0,
  y: 0,
  fontSize: "1rem",
  fontWeight: "normal",
}

export const setToolTip = (a, b, c) => {
  const w = globalThis.window && document.documentElement.clientWidth
  if (w <= 480) {
    return a
  } else if (w <= 768) {
    return b
  } else {
    return c
  }
}
