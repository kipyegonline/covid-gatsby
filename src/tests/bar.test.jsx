import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Bar, { ToolBar, Para } from "../components/covidreact/bar";
import { ToolTip } from "../components/svg/svg";

Enzyme.configure({ adapter: new Adapter() });

const data = [
  { TotalConfirmed: 400, TotalRecovered: 100, totalDeaths: 600 },
  { TotalConfirmed: 200, TotalRecovered: 140, totalDeaths: 1000 },
  { TotalConfirmed: 260, TotalRecovered: 300, totalDeaths: 750 },
  { TotalConfirmed: 140, TotalRecovered: 20, totalDeaths: 450 },
];
const btn = [
  { id: 1, selected: true, continent: "Africa", val: "africa" },
  { id: 2, selected: false, continent: "Asia", val: "asia" },
  { id: 3, selected: false, continent: "America", val: "america" },
  { id: 4, selected: false, continent: "Europe", val: "europe" },
];
describe("<Bar/> component", () => {
  // this is meant to fetch mocked value by the callback from the component
  const getValue = (data) => console.log("rec", data); // eslint-disable no-console
  const func = jest.fn(),
    regionFunc = jest.fn(),
    sortFunc = jest.fn();
  const wrapper = mount(
    <Bar
      data={data}
      a="Country"
      b="TotalConfirmed"
      btns={btn}
      sendBtn={func}
      sendRegion={regionFunc}
      addsort={sortFunc}
    />
  );

  describe(" Presence of UI elements", () => {
    it("renders", () => expect(toJSON(wrapper.find("svg"))).toMatchSnapshot());
    test("SVG presence and its props", () => {
      expect(wrapper.find("g").length).toBe(4);
      expect(wrapper.find("rect")).toHaveLength(4);
      expect(wrapper.find("button")).toHaveLength(5);
      expect(wrapper.find("div")).toHaveLength(3);

      expect(
        wrapper
          .find("button")
          .at(2)
          .hasClass("btn btn-md btn-info mr-2 blue lighten-1")
      ).toBe(true);
    });

    test("button click", () => {
      //onclick
      wrapper
        .find("button")
        .at(2)
        .simulate("click");
      //btn
      expect(
        wrapper
          .find("button")
          .at(2)
          .hasClass("btn btn-md btn-info mr-2 blue lighten-1")
      ).toBe(true);
      expect(func).toBeCalledTimes(1);
      expect(func).toBeCalledWith(2);
      expect(regionFunc).toBeCalledWith("asia");
    });

    // first btn on click
    wrapper
      .find("button")
      .at(0)
      .simulate("click");
    expect(
      wrapper
        .find("button")
        .at(0)
        .text()
    ).toBe("Unsort");
    // para
    expect(wrapper.find("p.comp").text()).toBe(
      "Kindly view this graph on computer or larger device for better user experience"
    );
  });

  describe("svg components", () => {
    test.skip("first rect and props", () => {
      const rect = wrapper.find("rect").at(0);
      rect.simulate("click", {
        target: {
          TotalConfirmed: 400,
          TotalRecovered: 100,
          totalDeaths: 600,
        },
      });
    });
  });

  describe("SVG guide", () => {
    setTimeout((done) => {
      const guide = wrapper.find("quickquide");
      expect(
        guide
          .find("p")
          .at(0)
          .text()
      ).toBe("Siii");
      done();
    });
  }, 2000);
});

describe("<ToolBar/>", () => {
  const data = {
    width: 100,
    Country: "Kenya",
    confirmed: 384,
    recovered: 129,
    deaths: 14,
    flag: "Ke.jpg",
  };
  const wrapper = shallow(<ToolBar tooltip data={data} />);

  it("receives the right props", () => {
    expect(wrapper.find("img").prop("src")).toBe("Ke.jpg");
    expect(
      wrapper
        .find(Para)
        .at(0)
        .text()
    ).toBe("Kenya");
    expect(
      wrapper
        .find(Para)
        .at(1)
        .text()
    ).toBe("Confirmed: 384");
  });
  expect(
    wrapper
      .find(Para)
      .at(2)
      .text()
  ).toBe("Deaths: 14");
  expect(
    wrapper
      .find(Para)
      .at(3)
      .text()
  ).toBe("Recovered: 129");
});
describe("<ToolTip/>", () => {
  const wrapper = shallow(
    <ToolTip tool data={{ dw: 100, bg: "red", x: 50, y: 50 }} />
  );
});
