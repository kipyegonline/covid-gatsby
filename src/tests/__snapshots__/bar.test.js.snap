// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar/> component  Presence of UI elements renders 1`] = `
<svg
  className="sc-AxjAm bJuYjR"
  height={300}
  width={300}
>
  <styled.g
    classlist=""
    gh={180}
    gw={200}
    x={80}
    y={40}
  >
    <g
      className="sc-AxirZ fKAEtI"
      height={180}
      width={200}
      x={80}
      y={40}
    >
      <styled.rect
        classList="rectsA"
        fill="blue"
        hc={[Function]}
        height={180}
        hh={[Function]}
        hl={[Function]}
        hover={0}
        key="0"
        opacity={1}
        tx={0}
        ty={0}
        width={133}
        x={34}
        y={0}
      >
        <rect
          className="sc-AxiKw iHLCYb"
          fill="blue"
          height={180}
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          width={133}
          x={34}
          y={0}
        />
      </styled.rect>
      <styled.rect
        classList="rectsA"
        fill="blue"
        hc={[Function]}
        height={90}
        hh={[Function]}
        hl={[Function]}
        hover={0}
        key="1"
        opacity={1}
        tx={0}
        ty={0}
        width={133}
        x={34}
        y={90}
      >
        <rect
          className="sc-AxiKw iHLCYb"
          fill="blue"
          height={90}
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          width={133}
          x={34}
          y={90}
        />
      </styled.rect>
      <styled.rect
        classList="rectsA"
        fill="blue"
        hc={[Function]}
        height={117}
        hh={[Function]}
        hl={[Function]}
        hover={0}
        key="2"
        opacity={1}
        tx={0}
        ty={0}
        width={133}
        x={34}
        y={63}
      >
        <rect
          className="sc-AxiKw iHLCYb"
          fill="blue"
          height={117}
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          width={133}
          x={34}
          y={63}
        />
      </styled.rect>
      <styled.rect
        classList="rectsA"
        fill="blue"
        hc={[Function]}
        height={63}
        hh={[Function]}
        hl={[Function]}
        hover={0}
        key="3"
        opacity={1}
        tx={0}
        ty={0}
        width={133}
        x={34}
        y={117}
      >
        <rect
          className="sc-AxiKw iHLCYb"
          fill="blue"
          height={63}
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          opacity={1}
          width={133}
          x={34}
          y={117}
        />
      </styled.rect>
      <TextArea
        fill="black"
        fontSize="1rem"
        fontWeight="bold"
        rotate={0}
        x={50}
        y={0}
      >
        <text
          fill="black"
          fontSize="1rem"
          fontWeight="bold"
          transform="rotate(0)"
          x={50}
          y={0}
        >
           - Corona Virus (COVID19) Cases -
           
          Mon May 11 2020
        </text>
      </TextArea>
      <TextArea
        fill="black"
        fontSize="1rem"
        fontWeight="bold"
        rotate={-90}
        x={-128.57142857142858}
        y={-50}
      >
        <text
          fill="black"
          fontSize="1rem"
          fontWeight="bold"
          transform="rotate(-90)"
          x={-128.57142857142858}
          y={-50}
        >
          Confirmed Cases
        </text>
      </TextArea>
      <styled.g
        classlist="ygrp"
        gh={0}
        gw={0}
        x={0}
        y={0}
      >
        <g
          className="sc-AxirZ jaeeck ygrp"
          height={0}
          width={0}
          x={0}
          y={0}
        />
      </styled.g>
      <styled.g
        classlist="xgrp"
        gh={0}
        gw={0}
        x={0}
        y={180}
      >
        <g
          className="sc-AxirZ fqhjsd xgrp"
          height={0}
          width={0}
          x={0}
          y={180}
        />
      </styled.g>
      <styled.g
        classlist="grid"
        gh={0}
        gw={0}
        x={0}
        y={0}
      >
        <g
          className="sc-AxirZ jaeeck grid"
          height={0}
          width={0}
          x={0}
          y={0}
        />
      </styled.g>
    </g>
  </styled.g>
</svg>
`;
